node/heartbeat/monitor.go

```diff
@@ -403,13 +415,11 @@ func (m *Monitor) computeInactiveHeartbeatMessages() {
 
 // GetHeartbeats returns the heartbeat status
 func (m *Monitor) GetHeartbeats() []PubKeyHeartbeat {
-	m.computeAllHeartbeatMessages()
-	m.computeInactiveHeartbeatMessages()
-
 	m.mutHeartbeatMessages.Lock()
 	status := make([]PubKeyHeartbeat, len(m.heartbeatMessages))
 	idx := 0
 	for k, v := range m.heartbeatMessages {
+		v.updateMutex.Lock()
```

Comment:
We use double locking (prone to deadlocks). m.mutHeartbeatMessages.Lock() and v.updateMutex.Lock()
Why do we need the newly added mutex?

Reply:
removed. works also without because the compute and get are locked by the initial mutex